Dart/src/com/jetbrains/lang/dart/ide/actions/DartPubActionBase.java

```diff
@@ -76,11 +76,19 @@ public static void setIsInProgress(boolean inProgress) {
   @Override
   public void update(@NotNull final AnActionEvent e) {
     //e.getPresentation().setText(getTitle());  "Pub: Build..." action name set in plugin.xml is different from its "Pub: Build" title
-    final boolean visible = getModuleAndPubspecYamlFile(e) != null;
+    final boolean visible = getModuleAndPubspecYamlFile(e) != null && !shouldSuppressForFlutter(e);
     e.getPresentation().setVisible(visible);
     e.getPresentation().setEnabled(visible && !isInProgress());
   }
 
+  private static boolean shouldSuppressForFlutter(@NotNull final AnActionEvent e) {
+    if (FlutterUtil.isFlutterPluginInstalled()) {
+      final Project project = e.getProject();
+      return project != null && FlutterUtil.hasFlutterModule(project);
```

Comment:
The same question: doesn't _current_ module type matter? There may be 2 modules in one project: one for Dart and one for Flutter, and this one may be a Dart one. Will flutter plugin handle it?